src/libs/ascent/runtimes/flow_filters/ascent_runtime_rover_filters.cpp

```diff
@@ -242,62 +252,33 @@ RoverXRay::execute()
   }
   std::shared_ptr<VTKHCollection> collection = data_object->as_vtkh_collection();
 
-  std::string absorption = params()["absorption"].as_string();
+  // Validate that the 'absorption' field exists in the dataset
+  std::string absorption = params()["rover/absorption"].as_string();
   if(!collection->has_field(absorption))
   {
     ASCENT_ERROR("Absorption field name '" << absorption << "' is not in the dataset");
   }
 
+  // Fetch the dataset associated with the 'absorption' field
   std::string topo_name = collection->field_topology(absorption);
-
-  // Returns an empty dataset if topo_name doesn't exist in the collection
   vtkh::DataSet &dataset = collection->dataset_by_topology(topo_name);
 
+  // Initialize rover and configure its behavior with the input params
   Rover rover;
   rover.update_settings(params());
 
   int mpi_comm_id = -1;
 #ifdef ASCENT_MPI_ENABLED
   mpi_comm_id = flow::Workspace::default_mpi_comm();
   rover::Logger::get_instance()->set_mpi_comm_id(mpi_comm_id);
-  // these use different styles of naming functions ....
-  rover::DataLogger::GetInstance()->set_mpi_comm_id(mpi_comm_id);
+  rover::DataLogger::get_instance()->set_mpi_comm_id(mpi_comm_id);
   rover.set_mpi_comm_handle(mpi_comm_id);
 #endif
-
-  if (params().has_path("precision"))
-  {
-    std::string precision = params()["precision"].as_string();
-    if (precision == "double")
-    {
-      rover.set_tracer_precision64();
-    }
-  }
   
-  // Adding a dataset to rover resets the camera bounds to the dataset bounds
+  // Adding a dataset to rover resets the camera bounds to the dataset bounds,
+  // but any camera params passed via the input params will take precedence
   rover.add_data_set(dataset);
-  // We then want to override the dataset-adjusted camera per the input params
-  rover.update_camera();
-
-  //
-  // Default render settings
-  //
-  RenderSettings render_settings;
-  render_settings.m_primary_field = absorption;
-
-  // TODO: investigate how/why this is getting set, even if emission is not specified
-  // example: if absorption == "radial", why is emission also == "radial"
-  if (params().has_path("emission"))
-  {
-    render_settings.m_secondary_field = params()["emission"].as_string();
-  }
-
-  if(params().has_path("unit_scalar"))
-  {
-    render_settings.m_energy_settings.m_unit_scalar = params()["unit_scalar"].to_float64();
-  }
-
-  rover.set_render_settings(render_settings);
+  // Calling execute initializes everything that rover needs based on the input params
```

Comment:
I guess `rover.add_data_set()` now handles all of the stuff that got removed here? It updates and camera and sets up render settings? I ask because we don't want any setup stuff happening in `execute()`.